arch/x86/core/intel64/irq.c

```diff
@@ -29,6 +29,28 @@ unsigned char _irq_to_interrupt_vector[CONFIG_MAX_IRQ_LINES];
 void (*x86_irq_funcs[NR_IRQ_VECTORS])(const void *);
 const void *x86_irq_args[NR_IRQ_VECTORS];
 
+#if defined(CONFIG_INTEL_VTD_ICTL) &&	\
+	!defined(CONFIG_INTEL_VTD_ICTL_XAPIC_PASSTHROUGH)
+
+#include <device.h>
+#include <drivers/interrupt_controller/intel_vtd.h>
+
+static const struct device *vtd;
+
+static bool get_vtd(void)
+{
+	if (vtd != NULL) {
+		return true;
+	}
+#define DT_DRV_COMPAT intel_vt_d
+	vtd = device_get_binding(DT_INST_LABEL(0));
+#undef DT_DRV_COMPAT
```

Comment:
Seems like most/all VT-D code should be moved to a dedicated c-file so you don't need to do this kind of hacks with `DT_DRV_COMPAT`. At least my impression is that `DT_DRV_COMPAT` is supposed to be declared in the beginning of c-files and applicable to the entire file.

Reply:
Actually, is there a way to avoid DT_DRV_COMPAT here? I get the usefulness of it in drivers/, but here when you need to get another device, what DTS macro should be used then?